core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -464,6 +464,10 @@ private[history] class FsHistoryProvider(conf: SparkConf, clock: Clock)
     secManager.checkUIViewPermissions(user)
   }
 
+  private def withListingLock[A](f: => A): A = listing.synchronized {
```

Comment:
No, why even bother with this method? seems like more code and indirection than just using `listing.synchronized {` below

Reply:
yes, simplified to `listing.synchronized`